pennylane/utils.py

```diff
@@ -436,10 +436,7 @@ def get_generator(op, return_matrix=False):
     This utility function should be removed once the aforementioned classes
     no longer differ in behaviour.
     """
-    gen = getattr(op, "generator", lambda: None)()
-
-    if gen is None:
-        raise ValueError(f"Operation {op.name} does not have a generator")
+    gen = op.generator()
```

Comment:
This is nicer!

Reply:
yep I am a fan! The generator being undefined is now treated at the source, and it is totally up to the developer if they want to catch it, or leave it and let the error propagate.

Reply:
Another nice artifact: it avoids the issue where _slightly different_ error messages are defined in differnent places in the code base. 

For example, in `master`, you have `Operation does not have a generator` in the metric tensor code, and `Generator not defined for Operation` in the Fourier module 😆